src/parser/core/modules/NormalisedEvents.ts

```diff
@@ -0,0 +1,178 @@
+import {DamageEvent, Event, HealEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import HitType from 'parser/core/modules/HitType'
+
+// Based on multi-hit margin previously in use for barrage and AOE modules
+const LEGACY_MUTLIHIT_DEDUPLICATION_TIME_WINDOW = 500
+
+const DAMAGE_EVENT_TYPES: Array<Event['type']> = [
+	'damage',
+	'calculateddamage',
+]
+
+const HEAL_EVENT_TYPES: Array<Event['type']> = [
+	'heal',
+	'calculatedheal',
+]
+
+export interface NormalisedDamageEvent extends Omit<DamageEvent, 'type'> {
+	type: 'normaliseddamage'
+	targetsHit: number
+	hits: number
+	ghostedHits: number
+	ghostedAmount: number
+	calculatedEvents: DamageEvent[]
+	confirmedEvents: DamageEvent[]
+	attachEvent(event: DamageEvent): void
+}
+
+export interface NormalisedHealEvent extends Omit<HealEvent, 'type'> {
+	type: 'normalisedheal'
+	targetsHit: number
+	hits: number
+	ghostedHits: number
+	ghostedAmount: number
+	calculatedEvents: HealEvent[]
+	confirmedEvents: HealEvent[]
+	attachEvent(event: HealEvent): void
+}
+
+export class NormalisedEvents extends Module {
+	static handle = 'normalisedEvents'
+
+	private static GenerateNewDamageEvent(event: DamageEvent): NormalisedDamageEvent {
+		return {
+			...event,
+			type: 'normaliseddamage',
+			targetsHit: 0,
+			hits: 0,
+			ghostedHits: 0,
+			ghostedAmount: 0,
+			calculatedEvents: [],
+			confirmedEvents: [],
+			attachEvent(event: DamageEvent): void {
+				if (event.type.includes('calculated')) {
+					this.calculatedEvents.push(event)
+				} else {
+					this.confirmedEvents.push(event)
+				}
+			},
+		}
+	}
+
+	private static GenerateNewHealEvent(event: HealEvent): NormalisedHealEvent {
+		return {
+			...event,
+			type: 'normalisedheal',
+			targetsHit: 0,
+			hits: 0,
+			ghostedHits: 0,
+			ghostedAmount: 0,
+			calculatedEvents: [],
+			confirmedEvents: [],
+			attachEvent(event: HealEvent): void {
+				if (event.type.includes('calculated')) {
+					this.calculatedEvents.push(event)
+				} else {
+					this.confirmedEvents.push(event)
+				}
+			},
+		}
+	}
+
+	@dependency private hitType!: HitType
+
+	_normalisedDamageEvents = new Map<number, NormalisedDamageEvent>()
+	_normalisedHealEvents = new Map<number, NormalisedHealEvent>()
+
+	normalise(events: Event[]): Event[] {
+		events.forEach((event) => {
+			if (DAMAGE_EVENT_TYPES.includes(event.type)) {
```

Comment:
Purely because of how many times this will be hit, it'd be cool to get the two const arrays defined as `Set`s, so you can use `.has`